drizzlepac/hapsequencer.py

```diff
@@ -812,3 +825,222 @@ def _get_envvar_switch(envvar_name, default=None):
+    # be no common WCS solution.
+    is_disjoint = grism_wcs_set.isdisjoint(direct_wcs_set)
+    if not is_disjoint:
+        # Generate the intersection between the Grism/Prism and direct images
+        grism_wcs_set &= direct_wcs_set
+        log.info("Common WCS solutions exist between the Grism/Prism and the direct images.")
+        log.info("    The common WCS solutions are: {}".format(grism_wcs_set))
+
+        # There is a preference for the active WCS for the viable images in the visit
+        # Check the grism_wcs_set for the preferential solutions
+        match_list = []
+        final_wcsname = ''
+        for wcs_item in wcs_preference:
+            match_list = fnmatch.filter(grism_wcs_set, wcs_item)
+            if match_list:
+                final_wcsname = match_list[0]
+                log.info("Final WCS solution to use for all Grism/Prism and direct images: {}".format(final_wcsname))
+                break
+
+        if final_wcsname:
+            # Finally, reset the primary WCS in all the images -- if the image is not in a skip list
+
+            for g_edp in tdp.grism_edp_list:
+                filename = g_edp.full_filename
+                if filename not in skip_grism_list:
+                    update_active_wcs(filename, final_wcsname)
+                    log.info("Setting the primary WCS for Grism/Prism image {} to {}.".format(filename, final_wcsname))
+
+                    # Only Grism/Prism images not in the skip list need to go into the manifest
+                    grism_product_list.append(filename)
+
+            for edp in tdp.edp_list:
+                filename = edp.full_filename
+                if filename not in skip_direct_list:
+                    update_active_wcs(filename, final_wcsname)
+                    log.info("Setting the primary WCS for direct image {} to {}.".format(filename, final_wcsname))
+
+    # Disjoint
+    else:
+        log.info("The Grism/Prism and direct images in this visit have no WCS solution common to both sets of data.")
+        log.info("The active WCS solution for each Grism/Prism and direct image is not changed.")
+
+    return grism_product_list
+
+# ------------------------------------------------------------------------------
+
+
+def update_active_wcs(filename, wcsname):
+    """
+    Utility to update the active/primary WCS solution
+
+    This small utility updates the active/primary WCS solution for the input
+    file with the WCS solution indicted by the input parameter "wcsname"
+
+    PARAMETERS
+    -----------
+    filename : str
+        Output SVM FLT/FLC filename
+
+    wcsname : str
+        Name of the desired WCS active/primary solution to be set for the filename
+    """
+    # For exposures with multiple science extensions (multiple chips),
+    # generate a combined WCS
+    num_sci_ext, extname = util.count_sci_extensions(filename)
+    extname_list = []
```

Comment:
Another way to create extname_list (lines 1040-1042) would be to use list comprehension with:

extname_list = [(extname, x+1) for x in range(num_sci_ext)]

Reply:
Cool - done.